`TestProxyKubeServerWatcher_StartsWithFaultyPrimary` flakiness

## Failure

#### Link(s) to logs

https://github.com/gravitational/teleport/actions/runs/29377332809/job/87233393643

#### Relevant snippet

```
=== FAIL: lib/kube/proxy/watcher TestProxyKubeServerWatcher_StartsWithFaultyPrimary (unknown)
2000/01/01 00:00:00 WARN Kube Server Watcher has failed. trace.component=kube_server_watcher error="creating watcher and waiting for init\n\tcreating a watcher\n\t\tcontext deadline exceeded" next_fallback_fetch=2000-01-01T00:00:00Z
    watcher_test.go:179: 
        	Error Trace:	/__w/teleport/teleport/lib/kube/proxy/watcher/watcher_test.go:179
        	Error:      	Should be true
        	Test:       	TestProxyKubeServerWatcher_StartsWithFaultyPrimary
        	Messages:   	Watcher starts healthy
panic: send on closed channel
```

--- a/lib/kube/proxy/watcher/watcher_test.go
+++ b/lib/kube/proxy/watcher/watcher_test.go
@@ -149,9 +149,11 @@ func testProxyKubeServerWatcherStartsWithFaultyPrimarySynctest(t *testing.T) {
 	fallback := &mockKubeServerWatcherGetter{}
 
 	var calls int32
+	primaryReady := make(chan time.Time)
 
 	primary.On("NewWatcher", mock.Anything, mock.Anything).
 		Return(nil, context.DeadlineExceeded).
+		WaitUntil(primaryReady).
 		Run(func(args mock.Arguments) {
 			atomic.AddInt32(&calls, 1)
 		})
@@ -169,16 +171,17 @@ func testProxyKubeServerWatcherStartsWithFaultyPrimarySynctest(t *testing.T) {
 	t.Cleanup(w.Close)
 
 	waitCh := make(chan error)
-	defer close(waitCh)
 
 	go func() {
 		// This should block until the watcher is ready or closed.
 		waitCh <- w.WaitInitialization()
 	}()
+	synctest.Wait()
 
 	require.True(t, isHealthy(w), "Watcher starts healthy")
 	require.False(t, w.IsInitialized())
 
+	close(primaryReady)
 	time.Sleep(2 * time.Second)
 	require.False(t, w.IsInitialized())
 	require.False(t, isHealthy(w), "Watcher should not be hot since primary is failing")

